[CS2113-T10-4] NUSFoodReviews

JoviYeung92 opened the pull request:
Allows users to leave/view reviews from various food stalls around NUS.

JoviYeung92 committed:
Merge branch 'master' into bryna-AboutUs

JoviYeung92 committed:
Merge pull request #3 from brynagoh/bryna-AboutUs

Update Bryna's Profile in AboutUs

JoviYeung92 committed:
Merge pull request #2 from kangxinwang/mybranch

edit about us file

nagiteja committed:
Merge pull request #1 from AY2021S2-CS2113-T10-4/master

to merge

nagiteja committed:
Add Nagi

nagiteja committed:
Merge pull request #5 from nagiteja/master

Add nagi

kangxinwang committed:
change class naming

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

brynagoh committed:
Rename Canteen class and fix checkstyle errors

brynagoh committed:
Merge pull request #7 from brynagoh/master

Rename Canteen class and fix checkstyle errors

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

nagiteja committed:
Merge pull request #2 from AY2021S2-CS2113-T10-4/master

merge with nagi

nagiteja committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp into master

* 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp:
  Rename Canteen class and fix checkstyle errors
  Add Nagi

Commit:
Update runtest.sh

JoviYeung92 committed:
try fix CI

brynagoh committed:
Merge pull request #1 from clarissegoh/master

Update runtest.sh

JoviYeung92 committed:
Merge pull request #8 from JoviYeung92/master

try fix CI

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

brynagoh committed:
Merge remote-tracking branch 'upstream/master'

kangxinwang committed:
tried fixing errors

kangxinwang committed:
Merge pull request #10 from kangxinwang/master

fixed errors

brynagoh committed:
Merge remote-tracking branch 'upstream/master'

brynagoh committed:
Add tests by adding args

Co-Authored-By: KevinNgWK <63909191+KevinNgWK@users.noreply.github.com>

brynagoh committed:
Fix checkstyle

brynagoh committed:
Fix linux/mac test arg

brynagoh committed:
Remove tests

brynagoh committed:
Merge pull request #9 from brynagoh/master

Update runtest.sh

KevinNgWK committed:
Merged upstream

KevinNgWK committed:
Resolved Merge Conflicts

KevinNgWK committed:
Added more OOP and some unit test

KevinNgWK committed:
Fixed checkstyle issues

KevinNgWK committed:
Fixed Unit Test for displayCommandTest and ExitCommandTest

KevinNgWK committed:
Added parser test

KevinNgWK committed:
Fixed checkstyle for parser test

brynagoh committed:
Merge pull request #17 from KevinNgWK/master

Added More OOP and some Unit Testing

JoviYeung92 committed:
test

JoviYeung92 committed:
merge

kangxinwang committed:
added read command

KevinNgWK committed:
First Attempt to fix CI issue

KevinNgWK committed:
Second Attempt to fix CI issue

brynagoh committed:
Add price and highlight menu for store, and related tests

- updates UserGuide for menu command
- adds exception messages to DukeExceptions
- updates storage to include menu prices

brynagoh committed:
Fix test imports for checkstyle

brynagoh committed:
Update ui tests, EXPECTED.txt

brynagoh committed:
Merge pull request #18 from KevinNgWK/master

First Attempt to fix CI issue

brynagoh committed:
Merge remote-tracking branch 'upstream/master'

JoviYeung92 committed:
test

kangxinwang committed:
Merge pull request #19 from brynagoh/master

Add price and highlight menu for store, and related tests

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	data/storage.txt
#	src/main/java/exceptions/DukeExceptions.java
#	src/main/java/parser/Parser.java
#	src/main/java/storage/Storage.java
#	src/main/java/stores/Store.java
#	src/main/java/ui/Ui.java

kangxinwang committed:
removed unused imports and enabled assertions

kangxinwang committed:
added read command for parsertest

kangxinwang committed:
fixed junit tests

kangxinwang committed:
Merge pull request #20 from kangxinwang/master

Add Read Command and removed unnecessary imports

JoviYeung92 committed:
merge

JoviYeung92 committed:
added admin verification and admin add store

JoviYeung92 committed:
Merge branch 'add_admin_verification_and_add_store'

nagiteja committed:
Merge pull request #3 from AY2021S2-CS2113-T10-4/master

Merge pull request from AY2021S2-CS2113-T10-4/tp

nagiteja committed:
Add add reviews command

nagiteja committed:
Merge pull request #22 from nagiteja/master

Add review and rating

nagiteja committed:
Merge pull request #23 from nagiteja/addreviewsandrating

Add add reviews command

JoviYeung92 committed:
fix runtest

KevinNgWK committed:
Revert "Add add reviews command"

KevinNgWK committed:
Merge pull request #24 from AY2021S2-CS2113-T10-4/revert-23-addreviewsandrating

Revert "Add add reviews command"

JoviYeung92 committed:
Merge pull request #21 from JoviYeung92/master

Added admin verification and admin add store

JoviYeung92 committed:
Revert "Added admin verification and admin add store"

kangxinwang committed:
Merge pull request #25 from AY2021S2-CS2113-T10-4/revert-21-master

Revert "Added admin verification and admin add store"

nagiteja committed:
Add "Add" command and passes all the tests

nagiteja committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp into master

* 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp:
  Revert "Added admin verification and admin add store"
  Revert "Add add reviews command"
  fix runtest
  added admin verification and admin add store
  test

# Conflicts:
#	src/main/java/command/AddCommand.java
#	src/main/java/parser/Parser.java
#	src/main/java/reviews/Review.java
#	src/test/java/command/AddCommandTest.java

nagiteja committed:
Merge pull request #26 from nagiteja/master

Add "Add" command and passes all the tests

JoviYeung92 committed:
added admin verification

JoviYeung92 committed:
added addStore for admin

JoviYeung92 committed:
fix runtest and checkstyle

JoviYeung92 committed:
Merge pull request #28 from JoviYeung92/master

added admin verification and addstore

JoviYeung92 committed:
added Junit

JoviYeung92 committed:
Merge pull request #29 from JoviYeung92/master

added Junit

KevinNgWK committed:
Added Saving Functionality

KevinNgWK committed:
Merge pull request #31 from KevinNgWK/master

Added Saving Functionality

kangxinwang committed:
update user guide

nagiteja committed:
Merge pull request #5 from AY2021S2-CS2113-T10-4/master

Merging with my base repo

nagiteja committed:
Merge branch 'master' of https://github.com/nagiteja/tp into master

* 'master' of https://github.com/nagiteja/tp:
  Added Saving Functionality
  added Junit
  fix runtest and checkstyle
  added addStore for admin
  added admin verification

brynagoh committed:
Fix menu keyword check to show correct exception message

kangxinwang committed:
add assertions

brynagoh committed:
Merge pull request #32 from brynagoh/master

Fix for menu exception message

kangxinwang committed:
add proper assertions

kangxinwang committed:
Merge pull request #33 from kangxinwang/assertions

Assertions

nagiteja committed:
Merge pull request #6 from AY2021S2-CS2113-T10-4/master

Merging with my repo

KevinNgWK committed:
Update user guide

nagiteja committed:
Changed AddCommand to AddReviewCommand

nagiteja committed:
Merge branch 'master' of https://github.com/nagiteja/tp into master

* 'master' of https://github.com/nagiteja/tp:
  add proper assertions
  add assertions
  Fix menu keyword check to show correct exception message
  update user guide

nagiteja committed:
Merge pull request #34 from nagiteja/master

Changed Addcommand to AddReviewCommand

JoviYeung92 committed:
add logo

KevinNgWK committed:
Merge remote-tracking branch 'upstream/master'

KevinNgWK committed:
Added assertions

KevinNgWK committed:
Merge pull request #36 from KevinNgWK/master

Added assertion and user user guide

JoviYeung92 committed:
add logo, fixed runtest

JoviYeung92 committed:
Merge pull request #35 from JoviYeung92/master

add logo

kangxinwang committed:
change assertions

JoviYeung92 committed:
adjust logo

JoviYeung92 committed:
restructure stores

JoviYeung92 committed:
restructure StoreMenu

JoviYeung92 committed:
restructure add reviews, read reviews and list

JoviYeung92 committed:
restructure exit command

JoviYeung92 committed:
v1.0 final touch up

JoviYeung92 committed:
Merge branch 'restructure'

JoviYeung92 committed:
Merge pull request #46 from JoviYeung92/master

V1.0 final touch up

brynagoh committed:
Refactor user commands

- allow multiple canteens
- store static variables to keep track of current canteen and store

brynagoh committed:
Refactor admin commands

- structure is slightly different from user (can look at standardising in the future)

brynagoh committed:
Merge pull request #47 from brynagoh/master

Refactor code for better OOP and canteens addition

KevinNgWK committed:
clean up checkUser

KevinNgWK committed:
Changed some comment

KevinNgWK committed:
Merge remote-tracking branch 'upstream/master'

kangxinwang committed:
update developer guide

KevinNgWK committed:
Added HomeCommand

kangxinwang committed:
Merge pull request #48 from kangxinwang/master

update developer guide

KevinNgWK committed:
Fixed Checkstyle

KevinNgWK committed:
Merge pull request #49 from KevinNgWK/master

Added HomeCommand and some refactoring

JoviYeung92 committed:
assertions

JoviYeung92 committed:
Merge pull request #50 from JoviYeung92/master

assertions

JoviYeung92 committed:
added test

JoviYeung92 committed:
Merge pull request #51 from JoviYeung92/master

added test

JoviYeung92 committed:
added DG project description

JoviYeung92 committed:
Merge pull request #52 from JoviYeung92/master

added DG project description

JoviYeung92 committed:
added more DG project description

JoviYeung92 committed:
Merge pull request #53 from JoviYeung92/master

added more DG project description

KevinNgWK committed:
Added cancellation when adding review

KevinNgWK committed:
Fixed checkstyle

KevinNgWK committed:
Merge pull request #54 from KevinNgWK/master

Added cancellation when adding reviews

kangxinwang committed:
add delete stores and delete reviews for admin side

kangxinwang committed:
fix checkstyle

kangxinwang committed:
fixed formatting

kangxinwang committed:
fixed number issue

brynagoh committed:
Merge pull request #55 from kangxinwang/master

add delete stores and delete reviews for admin side

brynagoh committed:
Add JUnit Tests for add/delete canteen, reset store

brynagoh committed:
Add assertions and logging

- Logs canteen/store creation

brynagoh committed:
Fix admin UI

- Delete canteen message
- Add store UI when canteen has no stores yet

brynagoh committed:
Add passing of Ui object instead of static methods

brynagoh committed:
Update Developer Guide for admin add canteen

brynagoh committed:
Update UserGuide for guest user commands

brynagoh committed:
Add fixes for checkstyle

brynagoh committed:
Merge pull request #56 from brynagoh/master

Add tests, assertions, developer/user guide updates, ui fixes

nagiteja committed:
Merge pull request #7 from AY2021S2-CS2113-T10-4/master

Merging with Nagi's tp repo

nagiteja committed:
Add Help Command

KevinNgWK committed:
Merge pull request #57 from nagiteja/master

Add Help Command

kangxinwang committed:
added sequence diagrams

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

kangxinwang committed:
Merge pull request #58 from kangxinwang/master

add sequence diagrams

kangxinwang committed:
add sequence diagram pictures

kangxinwang committed:
Merge pull request #59 from kangxinwang/master

add sequence diagram pictures

e0699194 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+* Display all the stores of the selected canteen: `list`
+* Exiting the application: `exit`
+
+#### Sequence Diagram for `admin`
```
Hi there, the sequence diagram seems to be missing!
![image](https://user-images.githubusercontent.com/77775515/113081506-61eab780-920b-11eb-8f22-5d670d94ed4d.png)

e0699194 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+the password. Then it will check the input against the set password. If fails then the user have to enter again or enter
+`exit` to exit the application.
+
+###[Proposed] delete feature 
```
Hi there! It seems that your text may be incorrectly formatted. It could be due to the missing space after ###.

![image](https://user-images.githubusercontent.com/77775515/113081651-aaa27080-920b-11eb-8583-95c4d75e5036.png)

NgManSing replied:
Should the format of this line as follows?
```
### [Proposed] delete feature 
```

yanli1215 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+* Exiting the application: `exit`
+
+#### Sequence Diagram for `admin`
+When the user enters `2` to go into the admin page, the AdminVerification() is called. It will ask the user to input 
```
Not enough visuals: lack of sequence diagram

e0699194 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+it once they logged in as admin. Admin has the ability to add/delete canteen, store and review. NUSFOODREVIEW has a storage file
+that allows the saving and loading of data. This allows all parameters of the data to be stored and read whenever is needed.
+
+##Implementation
+Below is the list of commands available.
```
Hi there! Just another formatting error!
![image](https://user-images.githubusercontent.com/77775515/113081979-4502b400-920c-11eb-8a55-ac8b572653ba.png)

yanli1215 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+###[Admin] Add Canteen
+####Implementation
+
+When the application is launched, a Ui object and Parser object is instantiated.
```
The explanation is unnecessarily long. A diagram can be used to show the implementation

e0699194 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+## Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Project overview
```
Hi there! The project overview seems a little wordy. Might I suggest breaking it up to a few chunks ?
![image](https://user-images.githubusercontent.com/77775515/113082205-9ca11f80-920c-11eb-923f-f7f4af737d41.png)

NgManSing replied:
Is it better to add more pictures or represent ideas in point form so that your overview could be less wordy?

yanli1215 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+`exit` to exit the application.
+
+###[Proposed] delete feature 
+####Proposed Implementation
```
the text format seems not correct here. need an empty line in markdown to show the new line

yanli1215 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+A switch class is then used to determine the command to instantiate. 
+To add a new canteen, the user must enter '2'. 
+
+Ui#showAddCanteen() is called to display the add canteen prompt.
```
I think the code part like showAddCanteen() can be surrounded by `` in stead of # to make the code more clear.

NgManSing reviewed (commented):
I think the content is OK~

NgManSing commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
 # Developer Guide
```
Is it better to have more visuals like sequence diagrams for better illustration?

NgManSing commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,88 @@
+
+Step 3. The user selects a canteen from the canteen list and canteen is deleted.
+
+###[Admin] Add Canteen
+####Implementation
```
Should the format of this line as follows?
```
### [Admin] Add Canteen
#### Implementation
```

NgManSing reviewed (commented):
I think the content is OK. It would be better if there could be more visuals.

kangxinwang committed:
add seqeunce diagrams to developer guide

kangxinwang committed:
Merge pull request #60 from kangxinwang/master

add seqeunce diagrams to developer guide

KevinNgWK committed:
Minor updates to developer guide

KevinNgWK committed:
Merge remote-tracking branch 'upstream/master'

brynagoh committed:
Update DeveloperGuide for add/delete canteen, displaymenus

- Added sequence diagrams for AddCanteen, DeleteCanteen, DisplayMenus, Main and Admin
- Updated DeveloperGuide for AddCanteen, DeleteCanteen, DisplayMenus

brynagoh committed:
Merge pull request #61 from brynagoh/master

Update DeveloperGuide for add/delete canteen, displaymenus

JoviYeung92 committed:
read data

JoviYeung92 committed:
read data checkstyle fixed

kangxinwang committed:
add images to developer's guide

JoviYeung92 committed:
read data checkstyle fixed

JoviYeung92 committed:
Merge pull request #62 from JoviYeung92/admin_functions

read files

JoviYeung92 committed:
data file

JoviYeung92 committed:
Merge branch 'master' into admin_functions

KevinNgWK committed:
Merge remote-tracking branch 'upstream/master'

KevinNgWK committed:
Added HomeCommand test

KevinNgWK committed:
Updated DeveloperGuide

KevinNgWK committed:
Merge pull request #63 from KevinNgWK/master

Updated DeveloperGuide and added tests

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	docs/DeveloperGuide.md

kangxinwang committed:
rename test

kangxinwang committed:
rename test

kangxinwang committed:
merge

kangxinwang committed:
edit storage file

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	data/storage.txt
#	docs/DeveloperGuide.md

kangxinwang committed:
Merge pull request #64 from kangxinwang/master

edit developer's guide

kangxinwang committed:
remove merge errors

kangxinwang committed:
Merge pull request #65 from kangxinwang/master

remove merge errors

kangxinwang committed:
update developer's guide

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

kangxinwang committed:
update user guide

kangxinwang committed:
Merge pull request #66 from kangxinwang/master

update user guide

JoviYeung92 committed:
rework writing to file

JoviYeung92 committed:
rework writing to file

JoviYeung92 committed:
Merge branch 'master' into master

JoviYeung92 committed:
Merge pull request #67 from JoviYeung92/master

rework writing to file

JoviYeung92 committed:
merge

JoviYeung92 committed:
Updated DG with addStore

JoviYeung92 committed:
Updated DG with addStore

kangxinwang committed:
remove manifest

nagiteja committed:
Merge pull request #8 from AY2021S2-CS2113-T10-4/master

Merging with Nagi

kangxinwang committed:
update user guide and developer's guide

kangxinwang committed:
Merge pull request #68 from kangxinwang/master

update developer's guide and user's guide

JoviYeung92 committed:
restructure reading from file

JoviYeung92 committed:
restructure reading from file

JoviYeung92 committed:
Merge pull request #69 from JoviYeung92/master

restructure reading from file

JoviYeung92 committed:
1)add menu (for admin)
2)add review to file
3)add canteen to file
4)add store to file
5)add menu to file
5)all read function stable

JoviYeung92 committed:
Merge pull request #70 from JoviYeung92/write_to_file

Added the following functions

JoviYeung92 committed:
textfile update

JoviYeung92 committed:
Merge branch 'master' into write_to_file

nagiteja committed:
Merge pull request #9 from AY2021S2-CS2113-T10-4/master

Merging with nagi

nagiteja committed:
Merge branch 'master' of https://github.com/nagiteja/tp into master

* 'master' of https://github.com/nagiteja/tp:
  1)add menu (for admin) 2)add review to file 3)add canteen to file 4)add store to file 5)add menu to file 5)all read function stable
  restructure reading from file
  update user guide and developer's guide
  remove manifest
  Updated DG with addStore
  Updated DG with addStore
  merge

kangxinwang committed:
merge with master

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	data/storage.txt

kangxinwang committed:
edit delete command

kangxinwang committed:
fixed checkstyle

kangxinwang committed:
made changes to store about reviewCount

nagiteja committed:
Add Time Stamp

kangxinwang committed:
Merge pull request #72 from nagiteja/master

Add Time Stamp Feature

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	data/storage.txt

kangxinwang committed:
Merge pull request #71 from kangxinwang/master

merge and edit delete reviews command

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,196 @@
 # Developer Guide
 
-## Design & implementation
+## Design 
```
It would be good if we can have a class diagram to show the overall design of the system.

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,196 @@
+
+### [Public user] Display Menus Feature
+#### Implementation
+![DisplayMenus Sequence Diagram](./img/DisplayMenus.png)
```
It would be better if you could have an example of the input and output for the command.

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,196 @@
+
+### [Admin] Delete Canteen
+#### Implementation
+![DeleteCanteen Sequence Diagram](./img/DeleteCanteen.png)
```
I have a question for this function here: Does deleting the canteen delete the review of the canteen at the same time?

Mick609 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,196 @@
+A new Canteen object is then instantiated, and added to the ArrayList of canteens.
+`Ui#showAddCanteenSuccess()` is called to display canteen added confirmation.
+
+### [Admin] Add Store
```
In this system implementation, what is the difference between the canteen and the store?

JoviYeung92 committed:
Merge remote-tracking branch 'upstream/master'

JoviYeung92 committed:
added delete menu function

JoviYeung92 committed:
1)update file when delete Canteen
2)update file when delete Store
3)update file when delete review
4)update file when delete menu

JoviYeung92 committed:
Merge pull request #74 from JoviYeung92/master

Update write to file, add menu and delete menu

JoviYeung92 committed:
Update storage.txt

JoviYeung92 committed:
update UG

JoviYeung92 committed:
Merge branch 'master' into master

JoviYeung92 committed:
Merge pull request #76 from JoviYeung92/master

update UG

KevinNgWK committed:
Added Login functionality

KevinNgWK committed:
Added Login to HelpCommand

KevinNgWK committed:
Fix Checkstyle

KevinNgWK committed:
Added Cancellation to AddCanteen

KevinNgWK committed:
Added cancellation to AddStore

KevinNgWK committed:
Added cancellation to AddMenu

KevinNgWK committed:
Added cancellation for DeleteCanteen

KevinNgWK committed:
Updated tests for deleteCanteenCommand

KevinNgWK committed:
Fixed checkstyle

KevinNgWK committed:
Fixed tests

KevinNgWK committed:
Added Cancellation for DeleteStore

KevinNgWK committed:
Added Cancellation for DeleteMenu

KevinNgWK committed:
Handled testing datafiles

KevinNgWK committed:
Merge pull request #77 from KevinNgWK/master

Added LoginCommand and Cancellations

kangxinwang committed:
merge

kangxinwang committed:
update user guide

kangxinwang committed:
change formatting of user guide

kangxinwang committed:
Merge pull request #78 from kangxinwang/master

update user guide

kangxinwang committed:
add error message for rating beyond range

kangxinwang committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-T10-4/tp

# Conflicts:
#	data/storage.txt

kangxinwang committed:
fixed checkstyle

kangxinwang committed:
fixed num issue

kangxinwang committed:
Merge pull request #79 from kangxinwang/master

add error message for rating out of range

kangxinwang committed:
update path for user guide

kangxinwang committed:
Merge pull request #80 from kangxinwang/master

update path for user guide

kangxinwang committed:
fix some of the errors spotted by testers

kangxinwang committed:
fixed checkstyle

kangxinwang committed:
Merge pull request #128 from kangxinwang/master

fix some of the errors spotted by testers

KevinNgWK committed:
Fixed adding menus

KevinNgWK committed:
Fixed empty canteen issues for public user

KevinNgWK committed:
Merge pull request #129 from KevinNgWK/master

Fixed adding menus

KevinNgWK committed:
Fixed index issue for adding menu

KevinNgWK committed:
Merge pull request #130 from KevinNgWK/master

Fixed index issue for adding menu

KevinNgWK committed:
Fixed Add Menu

KevinNgWK committed:
Merge pull request #131 from KevinNgWK/master

Fixed Add Menu

JoviYeung92 committed:
Add files via upload
